src/helper/class-ee-site.php

```diff
@@ -514,21 +516,48 @@ protected function init_le( $site_url, $site_fs_path, $wildcard = false ) {
 	 *
 	 * @param string $site_url  Name of site
 	 * @param $wildcard         Wildcard cert required?
+	 * @param bool   $allow_le  Allow LetsEncrypt
 	 *
 	 * @return array
 	 */
-	private function get_cert_domains( string $site_url, $wildcard ): array {
+	private function get_cert_domains( string $site_url, $wildcard, $allow_le = false ): array {
 
 		$domains = [ $site_url ];
 		if ( $wildcard ) {
 			$domains[] = "*.{$site_url}";
 		} else {
-			$domains[] = $this->get_www_domain( $site_url );
+
+			if ( true === $allow_le ) {
+				$domains[] = $this->get_www_domain( $site_url );
+			}
 		}
 
 		return $domains;
 	}
 
+	/**
+	 * Check www is working with site domain.
+	 *
+	 * @return bool
+	 */
+	protected function check_www_subdomain():bool {
+		$random_string = random_password();
+		file_put_contents( $this->site_data['site_fs_path'] . '/app/src/ssl_check.php', $random_string );
+
+		$site_url = 'www.' . $this->site_data['site_url'] . '/ssl_check.php';
+		$curl     = curl_init();
+		curl_setopt( $curl, CURLOPT_URL, $site_url );
+		curl_setopt( $curl, CURLOPT_RETURNTRANSFER, true );
+		curl_setopt( $curl, CURLOPT_HEADER, false );
+		$data = curl_exec( $curl );
+		curl_close( $curl );
```

Comment:
Add cleanup of the created file as soon as the `curl` request is done and we have the results.